src/condor_tools/htcondor_cli/annex_create.py

```diff
@@ -0,0 +1,720 @@
+        f"-C {local_script_dir} {target}.multi-pilot",
+        f"-C {str(token_file.parent)} {token_file.name}",
+        f"-C {str(password_file.parent)} {password_file.name}",
+    ]
+    files = " ".join(files)
+
+    transfer_files(
+        logger,
+        ssh_connection_sharing,
+        ssh_target,
+        ssh_indirect_command,
+        remote_script_dir,
+        files,
+        "populate remote temporary directory",
+    )
+
+
+def transfer_files(
+    logger,
+    ssh_connection_sharing,
+    ssh_target,
+    ssh_indirect_command,
+    remote_script_dir,
+    files,
+    task,
+):
+    # FIXME: Pass an actual list to Popen
+    proc = subprocess.Popen(
+        [
+            f'tar -c -f- {files} | ssh {" ".join(ssh_connection_sharing)} {ssh_target} {" ".join(ssh_indirect_command)} tar -C {remote_script_dir} -x -f-'
+        ],
+        shell=True,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        text=True,
+    )
+
+    try:
+        out, err = proc.communicate(timeout=REMOTE_POPULATE_TIMEOUT)
+
+        if proc.returncode == 0:
+            return 0
+        else:
+            logger.error(f"Failed to {task}, aborting.")
+            logger.warning(f"stdout: {out.strip()}")
+            logger.warning(f"stderr: {err.strip()}")
+            raise RuntimeError(f"Failed to {task}")
+
+    except subprocess.TimeoutExpired:
+        logger.error(
+            f"Failed to {task} in {REMOTE_POPULATE_TIMEOUT} seconds, aborting."
+        )
+        proc.kill()
+        out, err = proc.communicate()
+        raise RuntimeError(f"Failed to {task} in {REMOTE_POPULATE_TIMEOUT} seconds")
+
+
+def extract_full_lines(buffer):
+    last_newline = buffer.rfind(b"\n")
+    if last_newline == -1:
+        return buffer, []
+
+    lines = [line.decode("utf-8") for line in buffer[:last_newline].split(b"\n")]
+    return buffer[last_newline + 1 :], lines
+
+
+def process_line(line, update_function):
+    control = "=-.-= "
+    if line.startswith(control):
+        command = line[len(control) :]
+        attribute, value = command.split(" ")
+        update_function(attribute, value)
+    else:
+        print("   ", line)
+
+
+def invoke_pilot_script(
+    ssh_connection_sharing,
+    ssh_target,
+    ssh_indirect_command,
+    remote_script_dir,
+    target,
+    annex_name,
+    queue_name,
+    collector,
+    token_file,
+    lifetime,
+    owners,
+    nodes,
+    allocation,
+    update_function,
+    cluster_id,
+    password_file,
+):
+    args = [
+        "ssh",
+        *ssh_connection_sharing,
+        ssh_target,
+        *ssh_indirect_command,
+        str(remote_script_dir / f"{target}.sh"),
+        annex_name,
+        queue_name,
+        collector,
+        str(remote_script_dir / token_file.name),
+        str(lifetime),
+        str(remote_script_dir / f"{target}.pilot"),
+        owners,
+        str(nodes),
+        str(remote_script_dir / f"{target}.multi-pilot"),
+        str(allocation),
```

Comment:
Didn't we decide that this had to be `f'{allocation}'` to behave properly if `allocation` were `None`?

Reply:
(To whit, `stampede.sh` depends on `allocation` showing up as the string `None` on the command-line to preserve parameter indices.)

Reply:
It's the same:
```python
>>> allocation = None
>>> str(allocation) == f"{allocation}"
True
>>> str(allocation)
'None'
```

Reply:
Sorry.  You'd actually told me that before and I forgot.